### PR TITLE
Provides /usage sub-map with all pools in the default case

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- adding `pools` sub-map to `/usage`, which contains usage info for all pools, when the pool is not explicitly provided

## Why are we making these changes?

We decided, as explained in issue #755, that it makes sense to default to providing the information for all pools, instead of just for the default pool.

### CODE REVIEW

**Comment by dposada** [comment]
@pschorf Thoughts on this?

File: scheduler/src/cook/mesos/api.clj [hunk 1]
```diff
@@ -2000,41 +2001,75 @@
...
-           :ungrouped (let [[job-uuids usage] (get breakdowns nil)]
-                        {:running-jobs job-uuids :usage (or usage zero-usage)})})))))
+        pool (get-in ctx [:request :query-params :pool])]
+    (if pool
+      (let [jobs (util/get-user-running-job-ents db user pool)]
+        (user-usage with-group-breakdown? jobs))
+      (let [jobs (util/get-user-running-job-ents db user)
+            default-pool-name (config/default-pool)
+            pool-name->jobs (group-by
+                              #(let [pool (:job/pool %)]
```
Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: Can you use `if-let` here?

File: scheduler/src/cook/mesos/util.clj [hunk 2]
```diff
@@ -471,23 +471,37 @@
...
-               :where
-               ;; Note: We're assuming that many users will have significantly more
-               ;; completed jobs than there are jobs currently running in the system.
-               ;; If not, we might want to swap these two constraints.
-               [?j :job/state :job.state/running]
-               [?j :job/user ?user]
-               [(cook.mesos.pool/check-pool $ ?j :job/pool ?pool-name ?requesting-default-pool)]]
-             db user pool-name' requesting-default-pool?)
-           (map (partial d/entity db))))))
+  ([db user]
```
Comment (pschorf) [inline scheduler/src/cook/mesos/util.clj]: Does this make sense to split into a separate function? I think intuitively a 2 argument version of this function would return the jobs in the default pool, not all jobs. In any case, we should update the docstring.

File: scheduler/test/cook/test/mesos/api.clj [hunk 3]
```diff
@@ -1632,3 +1636,34 @@
       (is (nil? (:job/name job-entity)))
       (is (= job-uuid (:uuid job-map-for-api)))
       (is (= "cookjob" (:name job-map-for-api))))))
+
+(deftest test-get-user-usage-no-usage
```
Comment (pschorf) [inline scheduler/test/cook/test/mesos/api.clj]: Can you add some more tests, for instance where they have usage in some pools but not others?

File: scheduler/src/cook/mesos/api.clj (hunk 1 above)
Comment (dposada) [reply scheduler/src/cook/mesos/api.clj]: Done, thanks

File: scheduler/src/cook/mesos/util.clj (hunk 2 above)
Comment (dposada) [reply scheduler/src/cook/mesos/util.clj]: Done

File: scheduler/test/cook/test/mesos/api.clj (hunk 3 above)
Comment (dposada) [reply scheduler/test/cook/test/mesos/api.clj]: Done

**Comment by dposada** [comment]
Can we merge this?

**Pull request merged**